src/components/Conversations/ConversationsMessages.tsx

```diff
@@ -0,0 +1,82 @@
+import { useRouter } from 'next/router';
+import { useState, useEffect } from 'react';
+import { serviceClient } from '~/utils/trpc';
+import { TextInput, Group, Button, Loader, Center, createStyles } from '@mantine/core';
+import { IconSend } from '@tabler/icons-react';
+import { ConversationsMessage } from './ConversationsMessage';
+
+export function ConversationMessages() {
+  const router = useRouter();
+  const { classes } = useStyles();
+  const [messages, setMessages] = useState<any>([]);
+  const [loading, setLoading] = useState(false);
+  const [text, setText] = useState<null | string>(null);
+
+  // TODO: Instead this should be a messages w/ pagination
+  useEffect(() => {
+    const fetchConversations = async () => {
+      setLoading(true);
+      const data = await serviceClient.messages.getMessagesByConversationId.query({
+        conversationId: router.query.conversationId || '',
+        first: 10,
+      });
+
+      setMessages(data.reverse());
+      setLoading(false);
+
+      return data;
+    };
```

Comment:
Looks like a good candidate for a hook. That way you can keep the business logic tied and should be easier to later start adding pagination functionality and whatnot